foundation-extensions/mobile-core/mobile-core-api-reference.md

```diff
@@ -1,1182 +1,1805 @@
+
+You can use the `setApplication` method to pass the Android `Application` instance to Mobile SDK.
+
+{% tabs %}
+{% tab title="Android" %}
+**Java**
+
+### setApplication
+
+**Syntax**
+
+```java
+public static void setApplication(final Application app)
+```
+
+**Example**
+
+```java
+public class CoreApp extends Application {
+
+   @Override
+   public void onCreate() {
+      super.onCreate();
+      MobileCore.setApplication(this);
+      MobileCore.start(null);
+   }
+}
+```
+{% endtab %}
+
+{% tab title="Xamarin" %}
+**C\#**
+
+### setApplication
+
+**Example**
+
+```csharp
+public class MainActivity : global::Xamarin.Forms.Platform.Android.FormsAppCompatActivity
+{
+  protected override void OnCreate(Bundle savedInstanceState)
+  {
+    base.OnCreate(savedInstanceState);
+    ACPCore.Application = this.Application;
+    ACPCore.Start(null);
+  }
+}
+```
+{% endtab %}
+{% endtabs %}
+
+
+
+
+
+
+
+
+## setLogLevel
+
+The logging APIs allow log messages to be tagged and filtered with the Mobile SDK log messages. This allows application developers to filter the logged messages based on the current logging mode.
+
+Application developers can use the `setLogLevel` method to filter the log messages that are coming from the Mobile SDK.
+
+From least to most verbose, the order of Mobile SDK logging modes is as follows:
+
+* ERROR
+* WARNING
+* DEBUG
+* VERBOSE
+
+When debugging on iOS, you can use `LogLevel.verbose` to enable all the logging messages that are coming from Mobile SDK and partner extensions. Similarly, on Android, you can use `LoggingMode.VERBOSE` to enable all the logging messages that are coming from Mobile SDK and partner extensions.
+
+In a production application, you should use a less verbose logging mode, such as error-level logging.
+
+By default, Mobile SDK logging mode is set to `LoggingMode.ERROR` for Android and `LogLevel.error`on iOS.
+
+{% hint style="info" %}
+* On **Android**, Mobile SDK uses the `android.util.Log` class to log messages.
+* On **iOS**, Mobile SDK uses `NSLog` to messages to Apple System Log facility.
+{% endhint %}
+
+{% tabs %}
+{% tab title="Android" %}
+**Java**
+
+### setLogLevel
+
+**Syntax**
+
+```java
+public static void setLogLevel(LoggingMode mode)
+```
+
+**Example**
+
+```java
+import com.adobe.marketing.mobile.LoggingMode;
+import com.adobe.marketing.mobile.MobileCore;
+
+MobileCore.setLogLevel(LoggingMode.VERBOSE);
+```
+{% endtab %}
 
 {% tab title="iOS \(AEP 3.x\)" %}
 **Swift**
 
-This method should be called to support the following use cases:
+#### setLogLevel
 
-* Tracking deep link clickthroughs
-  * From `application(_:didFinishLaunchingWithOptions:)`
-  * Extract `userInfo` from `url: UIApplication.LaunchOptionsKey`
-* Tracking push message clickthrough
-  * From `application(_:didReceiveRemoteNotification:fetchCompletionHandler:)`
+**Syntax**
 
-#### collectLaunchInfo
+```swift
+ public static func setLogLevel(_ level: LogLevel)
+```
+
+**Example**
+
+```swift
```

Comment:
Can we also show the import for the AEP 3.x tab? AEP 3.x requires AEPServices to be imported

Reply:
For setLogLevel you just need to import AEPCore. Should i still add that?

Reply:
I believe you still need AEPServices for [LogLevel](https://github.com/adobe/aepsdk-core-ios/blob/main/AEPServices/Sources/log/LogLevel.swift)